### PR TITLE
Update release instructions in README.md

### PR DESCRIPTION
**Pull request by ezio-melotti**
Follow-up of:
* #88 

This PR updates the release instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,22 +77,21 @@ $ docutils --writer=pseudoxml tests/fixtures/xpass/role-in-code-sample.rst
...
+2. Go on the [Releases page](https://github.com/sphinx-contrib/sphinx-lint/releases)
+3. Click "Draft a new release"
+4. Click "Choose a tag"
+5. Type the next vX.Y.Z version "Create new tag: vX.Y.Z on publish"
+6. Leave the "Release title" blank (it will be autofilled)
+7. Click "Generate release notes" and amend as required
+8. Click "Publish release"
+9. Check the tagged
+   [GitHub Actions build](https://github.com/sphinx-contrib/sphinx-lint/actions/workflows/deploy.yml)
+   has [deployed to PyPI](https://pypi.org/project/sphinx-lint/0.7.0/#history)
```
Comment (AlexWaygood) [inline README.md:94]:
```suggestion
   has [deployed to PyPI](https://pypi.org/project/sphinx-lint/#history)
```

File: README.md [hunk 2]
```diff
@@ -77,22 +77,21 @@ $ docutils --writer=pseudoxml tests/fixtures/xpass/role-in-code-sample.rst
...
-    git push
-    git push --tags
-
-To release on PyPI run:
-
-    python -m pip install --upgrade build twine
-    python -m build
-    python -m twine upload dist/*
+1. Make sure that the [CI tests pass](https://github.com/sphinx-contrib/sphinx-lint/actions)
+   and optionally double-check locally with friends projects by running:
```
Comment (AlexWaygood) [inline README.md:81]: not sure about the grammar here?

```suggestion
   and optionally double-check locally with friends' projects by running:
```

File: README.md (hunk 1 above)
Comment (ezio-melotti) [reply README.md]: Good catch!

File: README.md [hunk 3]
```diff
@@ -77,22 +77,21 @@ $ docutils --writer=pseudoxml tests/fixtures/xpass/role-in-code-sample.rst
...
-    python -m twine upload dist/*
+1. Make sure that the [CI tests pass](https://github.com/sphinx-contrib/sphinx-lint/actions)
+   and optionally double-check locally with "friends projects" by running:
+   
+       sh download-more-tests.sh
+       python -m pytest
+2. Go on the [Releases page](https://github.com/sphinx-contrib/sphinx-lint/releases)
+3. Click "Draft a new release"
+4. Click "Choose a tag"
+5. Type the next vX.Y.Z version "Create new tag: vX.Y.Z on publish"
```
Comment (AlexWaygood) [inline README.md:88]:
```suggestion
5. Type the next vX.Y.Z version into "Create new tag: vX.Y.Z on publish"
```

Comment (ezio-melotti) [inline README.md:88]:
```suggestion
5. Type the next vX.Y.Z version and select "Create new tag: vX.Y.Z on publish"
```

**Review by AlexWaygood — APPROVED** [review]
LGTM

**Pull request merged**